maps/mp/isnipe/extras/antihardscope.gsc

```diff
@@ -2,35 +2,57 @@
 #include maps\mp\_utility;
 #include maps\mp\gametypes\_hud_util;
 
-EnableAntiHardScope(time)
-{
-    self endon( "disconnect" );
-    self endon( "death" );
+/**
+ * Initialize the antiHardScope variables on the level.
+ *
+ * The following settings are available to be set outside this script:
+ * - adsTime (default: 3): the time that a player is allowed to aim down sight.
+ */
+init() {
+    if (!isDefined(level.antiHardScope)) {
+        level.antiHardScope = [];
+    }
+
+    // Set settings to default when they are not set.
+    _setSettingIfNotExists("adsTime", 3);
+}
 
-    if( !isDefined( time ) || time < 0.05 )
-        time = 3;
+/**
+ * Set defaultValue in level.antiHardScope[property] if it does not exist.
+ *
+ * @param property: the setting property to act upon
+ * @param defaultValue: the default value to set for the property if not set
+ */
+_setSettingIfNotExists(property, defaultValue) {
+    if (!isDefined(level.antiCamp[property])) {
+        level.antiCamp[property] = defaultValue;
+    }
+}
 
-    adsTime = 0;
+/**
+ * Define the antiHardScope behaviour each time a player is spawned.
+ */
+onPlayerSpawned() {
+    self endon("disconnect");
+    self endon("death");
 
-    for( ;; )
-    {
-        if( self playerAds() == 1 )
-            adsTime ++;
-        else
+    adsTime = 0;
+    stepTime = 0.05;
+    for(;;) {
+        if (self playerAds() == 1) {
+            adsTime += stepTime;
+        } else {
             adsTime = 0;
+        }
 
-        if( adsTime >= int( time / 0.05 ) )
-        {
+        if (adsTime >= level.antiHardScope["adsTime"]) {
             adsTime = 0;
-            self allowAds( false );
-
-            while( self playerAds() > 0 )
-                wait( 0.05 );
-
-            self allowAds( true );
+            self allowAds(false);
+            while(self playerAds() > 0) {
+                wait(stepTime);
+            }
+            self.allowAds(true);
```

Comment:
This throws:
`
******* script compile error *******
Error: bad syntax in file maps/mp/isnipe/extras/antihardscope.gsc, line 54:
            self.allowAds(true);
                        *
************************************`

Bad syntax. This should be `self allowAds(true);`